### DISCUSSION TITLE
'read_image' has been explicitly marked deprecated

### DISCUSSION THREAD

**Discussion by jmacey** [OpenImageIO/oiio#3902] [Q&A]
Hi all,

I'm just updating my teaching code for next year and I'm still using the methods for reading and writing in C++ from the docs here https://openimageio.readthedocs.io/en/latest/imageinput.html 

I now get warnings as below

```
imageio.h:1471:5: note: 'read_image' has been explicitly marked deprecated here
    OIIO_DEPRECATED("replace with version that takes subimage & miplevel parameters (2.0)")
```

What is the correct way to use the API, I don't seem to see any examples that use the newer version with subimage and miplevels.

Thanks

Jon

**Comment by lgritz** [comment]
If you're just trying to make a simple example, pass 0 for both the subimage and miplevel parameters.

Most ordinary image files have only one image in the file and are not MIP-mapped, so only rarely and for certain file formats would they ever take on other values. In VFX, multi-part exr files, and indexing a single image from video file, are the main (maybe close to only?) common cases where subimage > 0. And miplevel > 0 would only happen if you are examining the lower-res levels of a MIP map texture.

**Reply by jmacey** [reply]
Thanks for the reply Larry, I have updated my code to use ```in->read_image(0,0,0,-1,TypeDesc::UINT8, pixels.get());``` and all is working fine and my unit test pass.

**Comment by lgritz** [comment]
You're quite right about the examples in the docs still using the old API!

🎣 Mackerel!

Anybody feel like making a patch to fix this?
